backend/shared/src/helpers/user-contract-metrics.ts

```diff
@@ -1,26 +1,38 @@
-import * as admin from 'firebase-admin'
 import { groupBy } from 'lodash'
-
 import { Contract } from 'common/contract'
 import { Bet } from 'common/bet'
 import { calculateUserMetrics } from 'common/calculate-metrics'
-
-const firestore = admin.firestore()
+import { bulkUpsert } from 'shared/supabase/utils'
+import { createSupabaseDirectClient } from 'shared/supabase/init'
+import { ContractMetric } from 'common/contract-metric'
 
 export async function updateContractMetricsForUsers(
   contract: Contract,
   allContractBets: Bet[]
 ) {
-  const writer = firestore.bulkWriter()
-  const betsByUser = groupBy(allContractBets, 'userId')
+  const betsByUser = groupBy(allContractBets, (b) => b.userId)
+  const metrics: ContractMetric[] = []
+  for (const userId in betsByUser) {
+    const userBets = betsByUser[userId]
+    metrics.push(calculateUserMetrics(contract, userBets))
+  }
 
-  Object.entries(betsByUser).forEach(([userId, bets]) => {
-    const metrics = calculateUserMetrics(contract, bets)
-    writer.update(
-      firestore.collection(`users/${userId}/contract-metrics`).doc(contract.id),
-      metrics
-    )
-  })
+  await bulkUpdateContractMetrics(metrics)
+}
 
-  await writer.flush()
+export async function bulkUpdateContractMetrics(metrics: ContractMetric[]) {
```

Comment:
<img width="363" alt="Screenshot 2023-07-12 at 8 53 18 AM" src="https://github.com/manifoldmarkets/manifold/assets/23196210/b62a145c-c3d6-48d0-8d5f-bbd43d1855e6">
It looks like the table has many more columns than you're updating here, why leave them out?

Reply:
I'm leaving the on insert trigger that copies to those fields from `data`. which is how the data got there from the  firebase replicator.

This is the minimal change necessary for migrating at all. Moving read/writes to native columns can be easily done later